hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java

```diff
@@ -454,19 +442,23 @@ public BasicKeyInfo next() {
     }
   }
 
-  private FileStatusAdapter toFileStatusAdapter(OzoneFileStatus status) {
+  private FileStatusAdapter toFileStatusAdapter(OzoneFileStatus status,
+      String owner, URI defaultUri, Path workingDir) {
+    OmKeyInfo keyInfo = status.getKeyInfo();
+    short replication = (short) keyInfo.getFactor().getNumber();
     return new FileStatusAdapter(
-        status.getLen(),
-        status.getPath(),
+        keyInfo.getDataSize(),
+        new Path(OZONE_URI_DELIMITER + keyInfo.getKeyName())
+            .makeQualified(defaultUri, workingDir),
         status.isDirectory(),
-        status.getReplication(),
+        replication,
         status.getBlockSize(),
-        status.getModificationTime(),
-        status.getAccessTime(),
-        status.getPermission().toShort(),
-        status.getOwner(),
-        status.getGroup(),
-        status.getPath(),
+        keyInfo.getModificationTime(),
+        keyInfo.getModificationTime(),
+        status.isDirectory() ? (short) 00777 : (short) 00666,
```

Comment:
How do we plan to map the dir/dir permission into Ozone ACLs? It is currently hard coded to 777 for dir and 666 for file.